### PR TITLE
Allow to lint a file glob instead of all files

### PR DESCRIPTION
**Pull request by bradfrizzell**
`gulp lint` runs across all files. Allow to run linter against just a file glob
`gulp lint --files=<file glob>`

### CODE REVIEW

File: build-system/tasks/lint.js [hunk 1]
```diff
@@ -127,7 +127,7 @@ function lint() {
   if (argv.fix) {
     options.fix = true;
   }
-  const stream = initializeStream(config.lintGlobs, {});
+  const stream = initializeStream(argv.file || config.lintGlobs, {});
```
Comment (rsimha) [inline build-system/tasks/lint.js]: This needs to be documented at https://github.com/ampproject/amphtml/blob/master/contributing/DEVELOPING.md#build--test

**Comment by jonkeller** [comment]
Brad, if the requested changes are more than you have time for, free free
to assign to me if you want.

File: build-system/tasks/lint.js (hunk 1 above)
Comment (bradfrizzell) [reply build-system/tasks/lint.js]: Done

**Comment by bradfrizzell** [comment]
@rsimha  good to know on that eslint stuff, I use emacs and there's eslint extensions available. I'll have to try that out.

And no worries @jonkeller it was a quick doc change only, but thanks :)

File: build-system/tasks/lint.js (hunk 1 above)
Comment (rsimha) [inline build-system/tasks/lint.js]: @bradfrizzell Sorry about not bringing this up earlier, but I just realized that it makes sense to change this from `argv.file` to `argv.files`, and allow developers to lint more than one file that they're editing in a branch, by specifying a list of files or a glob.

File: contributing/DEVELOPING.md [hunk 2]
```diff
@@ -47,6 +47,7 @@ The Quick Start Guide's  [One-time setup](getting-started-quick.md#one-time-setu
 | `gulp lint`                                                             | Validates against Google Closure Linter.                              |
 | `gulp lint --watch`                                                     | Watches for changes in files, Validates against Google Closure Linter.|
 | `gulp lint --fix`                                                       | Fixes simple lint warnings/errors automatically.                      |
+| `gulp lint --file=path/to/file.js`                                      | Lints just a single file.                                             |
```
Comment (rsimha) [inline contributing/DEVELOPING.md]: Per above comment, change this to:
```
gulp lint --files=<files-path-glob>
```
and
```
Lints just the files provided.
```

Comment (bradfrizzell) [reply contributing/DEVELOPING.md]: done

File: build-system/tasks/lint.js (hunk 1 above)
Comment (bradfrizzell) [reply build-system/tasks/lint.js]: done

File: contributing/DEVELOPING.md [hunk 3]
```diff
@@ -47,6 +47,7 @@ The Quick Start Guide's  [One-time setup](getting-started-quick.md#one-time-setu
 | `gulp lint`                                                             | Validates against Google Closure Linter.                              |
 | `gulp lint --watch`                                                     | Watches for changes in files, Validates against Google Closure Linter.|
 | `gulp lint --fix`                                                       | Fixes simple lint warnings/errors automatically.                      |
+| `gulp lint --files=path/to/file.js`                                     | Lints specified files.                                                |
```
Comment (rsimha) [inline contributing/DEVELOPING.md]: This should be `gulp lint --files=<files-path-glob>`

Comment (rsimha) [reply contributing/DEVELOPING.md]: And `Lints just the files provided.`

Comment (bradfrizzell) [reply contributing/DEVELOPING.md]: Whoops, fixed :)

Comment (rsimha) [reply contributing/DEVELOPING.md]: The PR description is out of date. Please update it before merging.

**Comment by bradfrizzell** [comment]
Tested the following: 

```
 gulp lint --files=extensions/amp-ad-network-doubleclick-impl/0.1/test/*
```
Files linted:
```
[13:39:43] Linting: /usr/local/google/home/bradfrizzell/A4A/amphtml/extensions/amp-ad-network-doubleclick-impl/0.1/test/test-amp-ad
[13:39:43] Linting: /usr/local/google/home/bradfrizzell/A4A/amphtml/extensions/amp-ad-network-doubleclick-impl/0.1/test/test-double
[13:39:43] Linting: /usr/local/google/home/bradfrizzell/A4A/amphtml/extensions/amp-ad-network-doubleclick-impl/0.1/test/test-double
[13:39:43] Linting: /usr/local/google/home/bradfrizzell/A4A/amphtml/extensions/amp-ad-network-doubleclick-impl/0.1/test/test-double
[13:39:43] Linting: /usr/local/google/home/bradfrizzell/A4A/amphtml/extensions/amp-ad-network-doubleclick-impl/0.1/test/test-double
[13:39:43] Linting: /usr/local/google/home/bradfrizzell/A4A/amphtml/extensions/amp-ad-network-doubleclick-impl/0.1/test/test-double
[13:39:43] Linting: /usr/local/google/home/bradfrizzell/A4A/amphtml/extensions/amp-ad-network-doubleclick-impl/0.1/test/test-double
[13:39:43] Success: No linter errors

```

AND

```
gulp lint --files=extensions/amp-ad-network-doubleclick-impl/0.1/test/test-amp-ad-network-doubleclick-impl.js
```
Files linted:
```
[13:40:09] Linting: /usr/local/google/home/bradfrizzell/A4A/amphtml/extensions/amp-ad-network-doubleclick-impl/0.1/test/test-amp-ad
[13:40:09] Success: No linter errors
```

**Comment by rsimha** [comment]
@bradfrizzell I just tried `gulp lint --files`, and it's breaking all the exceptions in `config.lintGlobs`. If you run it in a directory that contains things like `.html` or `.md` files, you're going to see lint errors from files that shouldn't be linted in the first place.

The correct way to implement this is to start with `config.lintGlobs`, and if `argv.files` is set, remove `**/*.js` from it and replace it with `argv.files`.

**Pull request merged**